### PR TITLE
Add unit tests for redeprovisioning

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

Changes proposed in this pull request:

- Add unit test for successfully redeprovision
- Add unit test for failed redeprovision

**Related issue(s)**

See also #2406

### CODE REVIEW

File: components/kyma-environment-broker/internal/provisioner/fake_client.go [hunk 1]
```diff
@@ -82,6 +82,19 @@ func (c *FakeClient) FindOperationByRuntimeIDAndType(runtimeID string, operation
...
 }
 
+func (c *FakeClient) FindLastOperationByRuntimeIDAndType(runtimeID string, operationType schema.OperationType) schema.OperationStatus {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var opStatus schema.OperationStatus
+	for _, status := range c.operations {
+		if *status.RuntimeID == runtimeID && status.Operation == operationType {
+			opStatus = status
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/provisioner/fake_client.go:92]: nit: if iterated from the end, you could break the cycle here and save a few CPU instructions :)

File: resources/kcp/charts/kyma-environment-broker/values.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ global:
   images:
     kyma_environment_broker:
       dir:
-      version: "PR-2555"
+      version: "PR-2570"
```
Comment (wozniakjan) [inline resources/kcp/charts/kyma-environment-broker/values.yaml:5]: nit: technically no need to bump for tests only

**Review by wozniakjan — APPROVED** [review]
/approve
/lgtm

**Comment by MarekMichali** [comment]
We can't assume that the last operation in the map will be in fact the most recent. We need to make sure in another way.

**Pull request merged**